src/controller/model/endpoint.ts

```diff
@@ -261,43 +264,62 @@ class Endpoint extends Entity {
         return this.pendingRequests.length > 0;
     }
 
-    public async sendPendingRequests(): Promise<void> {
-        while (this.pendingRequests.length > 0) {
-            const r = this.pendingRequests.shift();
-            try {
-                const result = await r.func();
-                r.resolve(result);
-            } catch (error) {
-                r.reject(error);
+    public async sendPendingRequests(fastPolling: boolean): Promise<void> {
+        const handled: PendingRequest[] = [];
+        for (const request of this.pendingRequests) {
+            if (fastPolling || request.queueSendWhen == 'active') {
+                try {
+                    const result = await request.func();
+                    request.resolve(result);
+                } catch (error) {
+                    request.reject(error);
+                }
```

Comment:
It would have been really neat if sendWhen active/fastpoll messages that failed could retry on next active/fastpoll period, instead of retrying immediately inside the adapter logic... But that's out of scope for this PR.